### PR TITLE
Adding support for automatic search indexing

### PR DESCRIPTION
**Pull request by noahmanger**
This adds Wagtail hooks for automatically adding, updating and deleting pages from the DigitalGov search engine. I basically took the same methods used for manually updating the index, refactored slightly and then added to Wagtail hooks on page creation, editing and deletion.

Still to do:
- [x] Add tests
- [x] Test with a dummy search engine (still waiting for a sandbox account to use)
- [x] Remove print statements

### CODE REVIEW

**Comment by noahmanger** [comment]
Updates: I just added tests and removed some of the print statements, though seems like it might be helpful to keep some of them in, unless there's a better way to log errors?

Also, still waiting on a sandbox account to test it with.

**Comment by noahmanger** [comment]
And it doesn't seem possible to include status messages in the Wagtail editor on success or error, unfortunately.

**Comment by noahmanger** [comment]
Alright this is all done. I was able to verify it works correctly with a sandbox search engine.

File: fec/fec/settings/base.py [hunk 1]
```diff
@@ -204,6 +204,8 @@
 FEC_DIGITALGOV_KEY = env.get_credential('FEC_DIGITALGOV_KEY')
 FEC_DIGITALGOV_DRAWER_KEY_MAIN = env.get_credential('DIGITALGOV_DRAWER_KEY_MAIN', '')
 FEC_DIGITALGOV_DRAWER_KEY_TRANSITION = env.get_credential('DIGITALGOV_DRAWER_KEY_TRANSITION', '')
+DIGITALGOV_BASE_API_URL = 'https://i14y.usa.gov/api/v1'
+DIGITALGOV_DRAWER_HANDLE = 'main'
```
Comment (ccostino) [inline fec/fec/settings/base.py]: Not a showstopper, but I'm wondering if these should also be pulled from environment variables too?

File: fec/search/tests/test_indexing.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    def test_update_document(self, scrape, m):
+        # It makes a PUT to the update endpoint
+        m.register_uri('PUT', 'http://localhost:3000/documents/123', status_code=200)
+        search.update_document(self.page)
+        self.assertTrue(m.called)
+
+
+    @requests_mock.Mocker()
+    def test_delete_document_in_prod(self, scrape, m):
+        # It makes a PUT to the update endpoint in production
```
Comment (ccostino) [inline fec/search/tests/test_indexing.py]: "It makes a `DELETE`..." :-)

File: fec/fec/settings/base.py (hunk 1 above)
Comment (noahmanger) [inline fec/fec/settings/base.py]: Yeah, probably makes sense, that way they're shared by the manual script.

Comment (ccostino) [inline fec/fec/settings/base.py]: I see the change below, but I meant these two values here specifically coming from environment variables too. :-)

Comment (noahmanger) [inline fec/fec/settings/base.py]: Ah I was confused. I think these can actually be hard-coded.

**Pull request merged**